pallets/gear/src/benchmarking/mod.rs

```diff
@@ -643,1295 +472,411 @@ benchmarks! {
-            ext_manager,
-            block_config,
-            context,
-            memory_pages,
-        } = prepare::<T>(instance.caller.into_origin(), HandleKind::Handle(ProgramId::from_origin(instance.addr)), vec![], 0u32.into())?;
-    }: {
-        core_processor::process::<
-            Externalities,
-            ExecutionEnvironment,
-        >(&block_config, context, memory_pages);
+        let mut res = None;
+        let exec = getter_bench::<T>("gr_origin", r)?;
+    }: {
+        res.replace(run_process(exec));
+    }
+    verify {
+        verify_process(res.unwrap());
     }
 
     gr_program_id {
         let r in 0 .. API_BENCHMARK_BATCHES;
-        let instance = Program::<T>::new(WasmModule::getter(
-            "env", "gr_program_id", r * API_BENCHMARK_BATCH_SIZE
-        ), vec![])?;
-        let Exec {
-            ext_manager,
-            block_config,
-            context,
-            memory_pages,
-        } = prepare::<T>(instance.caller.into_origin(), HandleKind::Handle(ProgramId::from_origin(instance.addr)), vec![], 0u32.into())?;
-    }: {
-        core_processor::process::<
-            Externalities,
-            ExecutionEnvironment,
-        >(&block_config, context, memory_pages);
+        let mut res = None;
+        let exec = getter_bench::<T>("gr_program_id", r)?;
+    }: {
+        res.replace(run_process(exec));
+    }
+    verify {
+        verify_process(res.unwrap());
     }
 
     gr_source {
         let r in 0 .. API_BENCHMARK_BATCHES;
-        let instance = Program::<T>::new(WasmModule::getter(
-            "env", "gr_source", r * API_BENCHMARK_BATCH_SIZE
-        ), vec![])?;
-
-        let Exec {
-            ext_manager,
-            block_config,
-            context,
-            memory_pages,
-        } = prepare::<T>(instance.caller.into_origin(), HandleKind::Handle(ProgramId::from_origin(instance.addr)), vec![], 0u32.into())?;
+        let mut res = None;
+        let exec = getter_bench::<T>("gr_source", r)?;
     }: {
-        core_processor::process::<
-            Externalities,
-            ExecutionEnvironment,
-        >(&block_config, context, memory_pages);
+        res.replace(run_process(exec));
+    }
+    verify {
+        verify_process(res.unwrap());
     }
 
     gr_value {
         let r in 0 .. API_BENCHMARK_BATCHES;
-        let instance = Program::<T>::new(WasmModule::getter(
-            "env", "gr_value", r * API_BENCHMARK_BATCH_SIZE
-        ), vec![])?;
-        let Exec {
-            ext_manager,
-            block_config,
-            context,
-            memory_pages,
-        } = prepare::<T>(instance.caller.into_origin(), HandleKind::Handle(ProgramId::from_origin(instance.addr)), vec![], 0u32.into())?;
-    }: {
-        core_processor::process::<
-            Externalities,
-            ExecutionEnvironment,
-        >(&block_config, context, memory_pages);
+        let mut res = None;
+        let exec = getter_bench::<T>("gr_value", r)?;
```

Comment:
```suggestion
        let exec = getter_bench::<T>(_, r)?;
```
this might work

Reply:
Unfortunately it does not work like this. Let's do something in future refactoring.
```
  error: in expressions, `_` can only be used on the left-hand side of an assignment
     --> /Users/gsobol/src/gear/pallets/gear/src/benchmarking/mod.rs:597:45
      |
  597 |         let exec = number_getter_bench::<T>(_, ValueType::I32, r)?;
      |                                             ^ `_` not allowed here

```